content/minutes/design-meeting/2019-11-09-unified-dataflow-framework.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Unify Dataflow Frameworks (#202)
+type: docs
+---
+
+# Design Meeting 2019-11-08
+
+[Zulip Stream](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/design.20meeting.202019-11-08)
+
+# Agenda
+
+- Design doc: https://hackmd.io/@39Qr_z9cQhasi25sGjmFnA/Skvd9rztS
+
+
+- Explain the current state of things
+    - We currently have two wholly separate APIs for dataflow, one for bit-vector problems and one that allows arbitrary transfer functions but still restricts the lattice to a powerset of indices.
+    - I want to merge these two to make code more DRY.
+    - The prototype implementation is described pretty thoroughly in the design doc, but maybe go over it a bit?
+
+- Discuss the prototype implementation in general
+    - The prototype is not very ambitious. It mostly preserves the existing `BitDenotation` API  but extends it to arbitrary transfer functions.
+    - Do we want to maintain the status quo? Have two separate implementations forever? Once arbitrary lattices are supported, maybe they won’t fit as nicely into the same API.
+        - OTOH, adding backwards analysis would have to be done in two places.
+    - There have been more ambitious changes attempted. Perhaps we want to do one of these instead? Or at least incorporate some of their ideas into a framework that hews closer to the existing one.
+        - @nagisa worked on a new dataflow framework that works for arbitrary lattices and transfer functions back in 2016.
+        - @eddyb was looking into a simultaneous forwards/backwards dataflow-like thing for NRVO (link?)
```

Comment:
Is the idea to add a link here?

Reply:
This was copy-pasted from the meeting agenda dropbox doc. I added markdown links.